overrides/cf.d.ts

```diff
@@ -289,97 +297,693 @@ interface RequestInitCfPropertiesImageMinify {
+
+  /** The certificate subject's distinguished name (legacy policies) */
+  certSubjectDNLegacy: string;
+
+  /** The certificate's serial number */
   certSerial: string;
-  certFingerprintSHA1: string;
-  /** "SUCCESS", "FAILED:reason", "NONE" */
-  certVerified: string;
-  certRevoked: string;
+
+  /** The certificate issuer's serial number */
+  certIssuerSerial: string;
+
+  /** The certificate's Subject Key Identifier */
   certSKI: string;
+
+  /** The certificate issuer's Subject Key Identifier */
+  certIssuerSKI: string;
+
+  /** The certificate's SHA-1 fingerprint */
+  certFingerprintSHA1: string;
+
+  /** The certificate's SHA-256 fingerprint */
+  certFingerprintSHA256: string;
+
+  /**
+   * The effective starting date of the certificate
+   *
+   * @example "Dec 22 19:39:00 2018 GMT"
+   */
+  certNotBefore: string;
+
+  /**
+   * The effective expiration date of the certificate
+   *
+   * @example "Dec 22 19:39:00 2018 GMT"
+   */
+  certNotAfter: string;
+};
+
+/** Placeholder values for TLS Client Authorization */
+type IncomingRequestCfPropertiesTLSClientAuthPlaceholder = {
+  certPresented: "0";
+  certVerified: CertVerificationStatus.Absent;
+  certRevoked: "0";
+  certIssuerDN: "";
+  certSubjectDN: "";
+  certIssuerDNRFC2253: "";
+  certSubjectDNRFC2253: "";
+  certIssuerDNLegacy: "";
+  certSubjectDNLegacy: "";
+  certSerial: "";
+  certIssuerSerial: "";
+  certSKI: "";
+  certIssuerSKI: "";
+  certFingerprintSHA1: "";
+  certFingerprintSHA256: "";
+  certNotBefore: "";
+  certNotAfter: "";
+};
+
+/** Enumeration of all outcomes of TLS verification */
+declare const enum CertVerificationStatus {
+  /** Authentication succeeded */
+  Success = "SUCCESS",
+
+  /** No certificate was presented */
+  Absent = "NONE",
+
+  /** Failed because the certificate was self-signed */
+  SelfSigned = "FAILED:self signed certificate",
+
+  /** Failed because the certificate failed a trust chain check */
+  Untrusted = "FAILED:unable to verify the first certificate",
+
+  /** Failed because the certificate not yet valid */
+  NotYetValid = "FAILED:certificate is not yet valid",
+
+  /** Failed because the certificate is expired */
+  Expired = "FAILED:certificate has expired",
+
+  /** Failed for another unspecified reason */
+  Failed = "FAILED",
+}
+
+/**
+ * An upstream endpoint's response to a TCP `keepalive` message from Cloudflare.
+ */
+declare const enum IncomingRequestCfPropertiesEdgeRequestKeepAliveStatus {
+  Unknown = 0,
+  /** no keepalives (not found) */
+  NoKeepAlives = 1,
+
+  /** no connection re-use, opening keepalive connection failed */
+  NoReuseOpenFailed = 2,
+
+  /** no connection re-use, keepalive accepted and saved */
+  NoReuseAccepted = 3,
+
+  /** connection re-use, refused by the origin server (`TCP FIN`) */
+  ReuseRefused = 4,
+
+  /** connection re-use, accepted by the origin server */
+  ReuseAccepted = 5,
+}
+
+/** ISO 3166-1 Alpha-2 codes */
+declare const enum Iso3166Alpha2Code {
```

Comment:
Is this definitely the list we use on the edge? Just because country names can be ...sensitive. What about just using the codes?